### PR TITLE
ftr: add json struct parser to support cli tool

### PR DESCRIPTION
**Pull request by LaurenceLiZhixin**
**What this PR does**:
add json interface parser tool
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
in json-struct-parser dir, I add parser to create interface from json file path, with test file.

**Does this PR introduce a user-facing change?**:

NONE

### CODE REVIEW

File: jparser/parser.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/buger/jsonparser"
```
Comment (AlexStocks) [inline jparser/parser.go:29]: pls move this pkg to the 2rd import block.

File: go.sum [hunk 2]
```diff
@@ -1,7 +1,12 @@
+github.com/apache/dubbo-go-hessian2 v1.8.0-rc1 h1:IJyHFLhJrNE5VdKtN6/0n7rJqCjHraHuORcHkZhwQyQ=
```
Comment (AlexStocks) [inline go.sum:1]: delete it pls.

File: go.sum [hunk 3]
```diff
@@ -1,7 +1,12 @@
...
+github.com/apache/dubbo-go-hessian2 v1.8.0-rc1/go.mod h1:7rEw9guWABQa6Aqb8HeZcsYPHsOS7XT1qtJvkmI6c5w=
+github.com/buger/jsonparser v1.0.0 h1:etJTGF5ESxjI0Ic2UaLQs2LQQpa8G9ykQScukbh4L8A=
+github.com/buger/jsonparser v1.0.0/go.mod h1:tgcrVJ81GPSF0mz+0nu1Xaz0fazGPrmmJfJtxjbHhUQ=
 github.com/davecgh/go-spew v1.1.0 h1:ZDRjVQ15GmhC3fiQ8ni8+OwkZQO4DARzQgrnXU1Liz8=
 github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
 github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c=
 github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
+github.com/dubbogo/gost v1.9.0/go.mod h1:pPTjVyoJan3aPxBPNUX0ADkXjPibLo+/Ib0/fADXSG8=
```
Comment (AlexStocks) [inline go.sum:9]: are you kidding me? using itself?

File: jparser/parser.go [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+func getJavaClassName(pkg interface{}) string {
+	val := reflect.ValueOf(pkg).Elem()
+	typ := reflect.TypeOf(pkg).Elem()
+	nums := val.NumField()
+	for i := 0; i < nums; i++ {
+		if typ.Field(i).Name == "JavaClassName" {
+			return val.Field(i).String()
+		}
+	}
+	fmt.Println("error: JavaClassName not found")
```
Comment (AlexStocks) [inline jparser/parser.go:229]: delete it.

**Comment by AlexStocks** [comment]
@LaurenceLiZhixin pls mkdir 'encoding/json' and move ur files to this dir.

**Review by AlexStocks — APPROVED** [review]

**Comment by LaurenceLiZhixin** [comment]
Fmt bugs above has been fixed!

File: encoding/json/parser.go [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+		case "time.Time":
+			userDefinedType = reflect.TypeOf(time.Time{})
+		case "float32":
+			userDefinedType = reflect.TypeOf(float32(0))
+		case "float64":
+			userDefinedType = reflect.TypeOf(float64(0))
+		case "bool":
+			userDefinedType = reflect.TypeOf(false)
+		default:
+			log.Println("warning: val", string(value), " in json is not supported")
```
Comment (watermelo) [inline encoding/json/parser.go:129]: Maybe need return a error.

File: encoding/json/parser.go [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+				break
+			}
+			v.Field(i).SetFloat(0)
+		case reflect.TypeOf(false):
+			if valStr == "true" || valStr == "1" {
+				v.Field(i).SetBool(true)
+			}
+
+		default:
+			log.Println("warning val: ", valStr, " in value is not supported")
```
Comment (watermelo) [inline encoding/json/parser.go:185]: Maybe need return a error.

**Review by watermelo — APPROVED** [review]

**Review by cityiron — APPROVED** [review]

File: encoding/json/parser.go (hunk 6 above)
Comment (LaurenceLiZhixin) [reply encoding/json/parser.go]: fixed!

File: encoding/json/parser.go [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+	perrors "github.com/pkg/errors"
+)
+
+// HessianRegisterPair define the pair to register to hessian
+type HessianRegisterPair struct {
+	JavaClassName string
+	Obj           interface{}
+}
+
+type jsonStructParser struct {
```
[the remark this answers is not in the document]
Comment (LaurenceLiZhixin) [reply encoding/json/parser.go]: fixed!

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**